source/guide_beehive.rst

```diff
@@ -0,0 +1,124 @@
+.. author:: momoaux <momoaux@koma666.de>
+
+.. tag:: web
+.. tag:: agent
+.. tag:: lang-go
+
+.. highlight:: console
+
+.. sidebar:: Logo
+
+  .. image:: _static/images/beehive.png
+      :align: center
+
+#####
+Beehive
+#####
+
+.. tag_list::
+
+Beehive is an event and agent system, which allows you to create your own
+agents that perform automated tasks triggered by events and filters. It is
+modular, flexible and really easy to extend for anyone. It has modules
+(we call them *Hives*), so it can interface with, talk to, or retrieve
+information from Twitter, Tumblr, Email, IRC, Jabber, RSS, Jenkins, Hue - to
+name just a few. Check out the full list of [available Hives](https://github.com/muesli/beehive/wiki/Available-Hives)
+in our Wiki.
+
+Connecting those modules with each other lets you create immensly useful agents.
+----
+
+.. note:: For this guide you should be familiar with the basic concepts of
+
+  * :manual:`Go <lang-go>`
+  * :manual:`Web Backends <web-backends>`
+  * :manual:`Supervisord <daemons-supervisord>`
+  * :manual:`Domains <web-domains>`
+
+License
+=======
+
+Beehive is released under the `AGPL-3.0 License`_.
+
+Prerequisites
+=============
+
+Nothing special
+.. warning:: But be careful, Beehive has no login and defaults to bind to localhost. So to connect you should use ssh tunnel or use any kind of .htaccess Basic Auth or similar.
+
+::
+
+
+Installation
+============
+
+Install ``Beehive``:
+
+.. code-block:: console
+
+  [isabell@stardust ~]$ cd
+  [isabell@stardust ~]$ mkdir beehive
+  [isabell@stardust ~]$ wget https://github.com/muesli/beehive/releases/download/v0.4.0/beehive_0.4.0_Linux_x86_64.tar.gz
+  [isabell@stardust ~]$ cd beehive
+  [isabell@stardust ~]$ tar -xzvf beehive_0.4.0_Linux_x86_64.tar.gz
+  [isabell@stardust ~]$ rm beehive_0.4.0_Linux_x86_64.tar.gz
+  
+
+Configuration
+=============
+
+Beehive will generate default config on startup
+
+Web Backend Config
+------------------
+
+.. warning:: Beehive_ will bind to locahost per default. There is no login. Please make sure to secure it before using it with external interface.
```

Comment:
How do you secure it? All lab guides should produce a at least reasonably secure setup, if you follow all of the instructions.

Reply:
It is secure because it binds to localhost. So no external connection possible but if started with -bind "0.0.0.0:8181" then you need to take care of access. is there a reverse proxy guide somewhere that I could link?

I don't know how to add htaccess on uberspace if directly binding web access to the service. If you can point me to the right direction, I'll get into it and update but I could not find something about securing services.

Reply:
Right now there is no way to add basic authentication to a web backend, if the app itself does not support it. If beehive has no authentication built-in, I guess the only way to run it securely is to only use it inside of SSH port forwarding.

Reply:
should I just remove the bind and add a little tutorial for Chrome socks proxy?

Reply:
or maybe just link here: https://www.chromium.org/developers/design-documents/network-stack/socks-proxy
and here https://securitymusings.com/article/462/how-to-set-up-a-socks-proxy-using-putty-ssh

Reply:
hm, I removed the sections that expose beehive to the internet. Still a helpful guide nonetheless! :)